woodwork/deserialize.py

```diff
@@ -57,6 +59,41 @@ def _typing_information_to_woodwork_table(table_typing_info, validate, **kwargs)
     kwargs = loading_info.get('params', {})
     table_type = loading_info.get('table_type', 'pandas')
 
+    logical_types = {}
+    semantic_tags = {}
+    column_descriptions = {}
+    column_metadata = {}
+    use_standard_tags = {}
+    dtypes = {}
+    for col in table_typing_info['column_typing_info']:
+        col_name = col['name']
+
+        ltype_metadata = col['logical_type']
+        ltype = ww.type_system.str_to_logical_type(ltype_metadata['type'], params=ltype_metadata['parameters'])
+
+        tags = col['semantic_tags']
+
+        if 'index' in tags:
+            tags.remove('index')
+        elif 'time_index' in tags:
+            tags.remove('time_index')
+
+        logical_types[col_name] = ltype
+        semantic_tags[col_name] = tags
+        column_descriptions[col_name] = col['description']
+        column_metadata[col_name] = col['metadata']
+        use_standard_tags[col_name] = col['use_standard_tags']
+
+        if col['physical_type']['type'] == 'category':
+            # Make sure categories are recreated properly
+            cat_values = col['physical_type']['cat_values']
+            cat_dtype = col['physical_type']['cat_dtype']
+            cat_object = pd.CategoricalDtype(pd.Series(cat_values, dtype=cat_dtype))
+            dtypes[col_name] = cat_object
+        elif not (ks and col['physical_type']['type'] == 'object'):
+            # Can't specify `object` for koalas
+            dtypes[col_name] = col['physical_type']['type']
```

Comment:
Why are we ignoring columns that indicate they've been stored with `'object'` dtype? This isn't possible in koalas, right? How will LatLong columns get handled? A user could have koalas installed but be using a Pandas DataFrame with LatLong columns, so maybe this needs to be looking at `table_type`?

I guess I'm not understanding what's happening here.

Reply:
Let me take another look at this. This might have been a change that became obsolete after later changes, and never got removed.

Reply:
Yeah, this can be removed. At one point I was setting the dtype for all columns in the dtypes dict, but then changed to doing it only for the category columns. This was necessary to handle the Koalas issues around the `object` dtype, but it no longer needed.

It has been removed!

Reply:
awesome!